src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala

```diff
@@ -36,14 +56,41 @@ object MissingLinkPlugin extends AutoPlugin {
 
       val cp = Attributed.data(fullClasspath.value)
       val classDir = (classDirectory in Compile).value
+      val failOnConflicts = missinglinkFailOnConflicts.value
+      val ignoreSourcePackages = missinglinkIgnoreSourcePackages.value
+      val ignoreDestinationPackages = missinglinkIgnoreDestinationPackages.value
       val conflicts = loadArtifactsAndCheckConflicts(cp, classDir, log)
+      val filteredConflicts =
+        filterConflicts(conflicts, ignoreSourcePackages, ignoreDestinationPackages, log)
+
+      if (filteredConflicts.nonEmpty) {
+        val initialTotal = conflicts.length
+        val filteredTotal = filteredConflicts.length
 
-      outputConflicts(conflicts, log)
-      if (conflicts.nonEmpty)
-        throw new MessageOnlyException("there were conflicts")
+        val diffMessage = if (initialTotal != filteredTotal) {
+          s"($initialTotal conflicts were found before applying filters)"
+        } else {
+          ""
+        }
+
+        log.warn(s"$filteredTotal conflicts found! $diffMessage")
+
+        outputConflicts(filteredConflicts, log)
+
+        if (failOnConflicts)
+          throw new MessageOnlyException(s"There were ($filteredTotal) conflicts")
```

Comment:
I'm not sure why the parens? Consider:
```suggestion
          throw new MessageOnlyException(s"There were $filteredTotal conflicts")
```